vault/identity_store_util.go

```diff
@@ -66,6 +73,18 @@ func (i *IdentityStore) loadGroups(ctx context.Context) error {
 				continue
 			}
 
+			// Ensure that there are no groups with duplicate names
+			groupByName, err := i.MemDBGroupByName(ctx, group.Name, false)
+			if err != nil {
+				return err
+			}
+			if groupByName != nil && !i.core.disableCaseInsensitiveIdentityNames {
+				return fmt.Errorf(`duplicate group names %q and %q;
+identity names are treated case insensitively unless
+'disable_case_insensitive_identity_names' config is set`,
+					group.NameRaw, groupByName.NameRaw)
+			}
```

Comment:
This is likely to happen during upgrade right? Would this halt startup?

Reply:
This worries me a bit, I wonder if there is a better escape hatch we can offer. It seems to me it's likely that many existing installations will hit this issue and need to disable case insensitivity, which defeats the purpose of making it case sensitive.

Reply:
Yes this would happen during upgrade. The goal was to switch over all identity names to be case insensitive for good. It would still work seemlessly for those who are using case sensitive names. The conflict would only arise if there are duplicate names with different casings for entities, groups and/or respective aliases. In that case, they will have to set this flag to continue operating as they would. Or, they could set this flag, remove the duplicate items, and reset the flag to get along with the regular flow.

Reply:
This should get a mention in the upgrade notes.

Reply:
I think if disable is set, we should detect and warn. That way people can set the flag, start up, see if anything is in conflict, take appropriate action.

Reply:
There is no flag now. So, the above discussion is not required.